File: frontend/src/types/floorplan.ts
export interface Room {
  id: string;
  name: string;
  type: 'bedroom' | 'bathroom' | 'kitchen' | 'living_room' | 'dining' | 'other';
  areaSqFt?: number;
  dimensions?: {
    width: number;
    length: number;
  };
}

export interface Floorplan {
  id: string;
  projectId: string;
  totalAreaSqFt: number;
  rooms: Room[];
  formatUrls: {
    svg?: string;
    dxf?: string;
    model3d?: string;
  };
  createdAt: Date;
}

File: backend/models/floorplan.py
"""
Floorplan specification model for AI FloorPlanner Backend.

Stub model for future CAD generation. Represents the output
of the Planning Agent — a structured architectural specification.
"""

from typing import Optional

from pydantic import BaseModel, Field


class RoomSpec(BaseModel):
    """Specification for a single room in the floor plan."""
    name: str
    room_type: str  # bedroom, bathroom, kitchen, living, etc.
    area_sqft: Optional[float] = None
    width: Optional[float] = None
    length: Optional[float] = None
    floor: int = 0


class FloorplanSpec(BaseModel):
    """
    Structured architectural specification produced by the Planning Agent.
    This is the input to the CAD Generation Agent.
    """
    rooms: list[RoomSpec] = Field(default_factory=list)
    stairs: Optional[dict] = None
    parking: Optional[dict] = None
    setbacks: Optional[dict] = None
    total_built_area_sqft: Optional[float] = None
    metadata: dict = Field(default_factory=dict)
